src/interface/file.rs

```diff
@@ -201,6 +201,34 @@ impl EmulatedFile {
         Ok(bytes_written)
     }
 
+    pub fn write_vectored_at(
+        &mut self,
+        bufs: &[IoSlice<'_>],
+        offset: usize,
+    ) -> std::io::Result<usize> {
+        let mut total_bytes_written = 0; //to keep track of the total number of bytes written.
+
+        if let Some(f) = &self.fobj {
+            // checks if the file object (fobj) exists.
+            let mut file = f.lock();
+            // Seek to the specified offset from the beginning of the file
+            file.seek(SeekFrom::Start(offset as u64))?; 
+            // moves the file pointer to the desired starting position (offset) from the
+            // beginning of the file.
+
+            // Use write_vectored for efficient writing from multiple buffers
+            total_bytes_written = file.write_vectored(bufs)?; 
```

Comment:
You should be able to use "write_vectored_at" here instead of seeking and using write_vectored